src/item.cpp

```diff
@@ -13216,7 +13217,11 @@ bool item::process_tool( Character *carrier, const tripoint &pos )
 
 bool item::process_blackpowder_fouling( Character *carrier )
 {
-    if( damage() < max_damage() && one_in( 2000 ) ) {
+    // rust is deterministic. 12 hours for first rust, then 24 (36 total), then 36 (72 total) and finally 48 (120 hours to go to XX)
+    // this speeds up by the amount the gun is dirty, 2-6x as fast depending on dirt level.
+    set_var( "rust_timer", get_var( "rust_timer", 0 ) + 1 + static_cast<int>( get_var( "dirt",
+             0 ) / 2000 ) );
+    if( damage() < max_damage() && get_var( "rust_timer", 0 ) > 43200 / ( damage() + 1 ) ) {
```

Comment:
`rust_timer` should be reset after increasing damage if it is not already, otherwise the gun will likely rust to XX in a few turns after 12 hours because `43200 / (damage() + 1)` will be lower every time the damage is increased.

Reply:
You're right, thanks for catching it.